benchmark/cmake/modules/AddSwiftBenchmarkSuite.cmake

```diff
@@ -200,6 +200,14 @@ function (swift_benchmark_compile_archopts)
       "-${BENCH_COMPILE_ARCHOPTS_OPT}"
       "-no-link-objc-runtime"
       "-I" "${srcdir}/utils/ObjectiveCTests")
+
+  if(SWIFT_BENCHMARK_GENERATE_OPT_VIEW)
+    if(NOT ${optflag} STREQUAL "Onone" AND "${bench_flags}" MATCHES "-whole-module.*")
+      set(common_options ${common_options} "-save-optimization-record")
+      set(optview_main_dir "${objdir}/opt-view")
```

Comment:
Can you initial this variable outside initially so that scope is communicated, i.e.:

```
set(optview_main_dir)
if (SWIFT_BENCHMARK_GENERATE_OPT_VIEW)
   ...
endif()
```